scripts/create-component/plop-templates-component/src/components/{{componentName}}/use{{componentName}}.ts.hbs

```diff
@@ -5,7 +5,7 @@ import { {{componentName}}Props, {{componentName}}State } from './{{componentNam
 /**
  * Consts listing which props are shorthand props.
  */
-export const {{camelCase componentName}}ShorthandProps = [];
+export const {{camelCase componentName}}ShorthandProps: (keyof {{componentName}}Props)[] = [];
```

Comment:
I don't mind this, but currently it's still 280+ properties and internally all the shorthand stuff doesn't really care and casts to either `GenericDictionary` or `any` anyway

Reply:
would be more useful to encourage an extra `ComponentSlots` interface and using that to type instead.

the props interface can then extend the interface, that's as far as you can go with type safety with the current state of things IMO

Reply:
I think we removed `GenericDictionary` from `ComponentProps` right? So at least this provides some validation against tyops.

I agree that we could further improve the type safety around slots. I like the idea of the `[Component]Slots` type, but we should probably run that idea by other people, perhaps with an RFC. I'll write one up.